### PR TITLE
Ignore `drop` on `&mut` references

### LINKED ISSUE #431: Panic when updating `RVec<RVec<f32>>`
The following code panic when trying to update `centers[i]`

```rust
#[flux::sig(fn(&RVec<f32>[@n], usize) -> RVec<f32>[n])]
fn normal(x: &RVec<f32>, w: usize) -> RVec<f32> {
    let mut i = 0;
    let mut res = RVec::new();
    while i < x.len() {
        res.push(x[i] / (w as f32));
        i += 1;
    }
    res
}

#[flux::sig(fn (n: usize, centers: &mut RVec<RVec<f32>[n]>[@k], new_centers: RVec<(usize, (RVec<f32>, usize))>))]
fn update_centers(
    n: usize,
    centers: &mut RVec<RVec<f32>>,
    new_centers: RVec<(usize, (RVec<f32>, usize))>,
) {
    for (i, (x, size)) in new_centers {
        centers[i] = normal(&x, size)
    }
}
```

This is the panic

```
error: internal compiler error: flux-refineck/src/type_env.rs:169:17: cannot move out of `*_21`, which is behind a `&mut` reference
  --> /home/nlehmann/phd/flux-rs/flux/attic/playground.rs:24:9
   |
24 |         centers[i] = normal(&x, size)
   |         ^^^^^^^^^^
-Ztrack-diagnostics: created at /rustc/700938c0781c9f135244bb1ec846fe1a5e03ae7d/compiler/rustc_errors/src/lib.rs:995:33
```

### PR DESCRIPTION
**Pull request by ranjitjhala**
Fixes #431 (?)

### CODE REVIEW

**Comment by ranjitjhala** [comment]
This isn't quite right ... it does something bizarre with the vtock generating a subtyping error of the form

```
note: incompatible types: `uninit` - `MapCell<λb0. <λb1. <λb2. C[b2] as Chip>::UserspaceKernelBoundary[b1] as UserspaceKernelBoundary>::StoredState[b0]>[]`
```

**Comment by ranjitjhala** [comment]
Ok @nilehmann, any thoughts on the hack above? It makes the `vtock` thing work (and our tests pass, I believe.)

File: crates/flux-refineck/src/type_env/place_ty.rs [hunk 1]
```diff
@@ -73,7 +73,7 @@ impl LookupKey for Path {
 pub(super) struct LookupResult<'a> {
     pub ty: Ty,
     pub is_strg: bool,
-    pub is_constant_index: bool, // doesn't feel right...
+    pub _is_constant_index: bool, // doesn't feel right...
```
Comment (nilehmann) [inline crates/flux-refineck/src/type_env/place_ty.rs:76]: Let's get rid of this

```suggestion
```

**Review by nilehmann — APPROVED** [review]
Let's hope I don't regret this

**Pull request merged**